### PR TITLE
fix: Improve trimming of Swift function names

### PR DESCRIPTION
**Pull request by Steelskin**
Some Swift for Windows function names were trimmed to incorrect tokens in the UI, resulting in hard-to-read stack traces. For instance, some functions were shown as `"throws"` and closures were shown as `"#1"`.

This change fixes the former issue by using the correct function name. Closures are now displayed as `"closure in <function name>"`, and initialization expressions are displayed as `"initializer expression of <value>"`, improving the user experience when browsing stack traces.

### Legal Boilerplate

Look, I get it. The entity doing business as "Sentry" was incorporated in the State of Delaware in 2015 as Functional Software, Inc. and is gonna need some rights from me in order to utilize my contributions in this here PR. So here's the deal: I retain all rights, title and interest in and to my contributions, and by keeping this boilerplate intact I confirm that Sentry can use, modify, copy, and redistribute my contributions, under Sentry's choice of terms.

### CODE REVIEW

File: tests/sentry/stacktraces/test_functions.py [hunk 1]
```diff
@@ -103,6 +103,18 @@
         ],
         ["pthread_cond_timedwait@@GLIBC_2.3.2", "pthread_cond_timedwait"],
         ["glob64@GLIBC_2.2", "glob64"],
+        [
+            "static WinUI.WindowsXamlManager.getForCurrentThread() throws -> WinUI.WindowsXamlManager?",
```
Comment (tristanlabelle) [inline tests/sentry/stacktraces/test_functions.py:107]: This example may be confusing to folks who don't expect to see Windows APIs in Swift code, but no biggie.

**Review by tristanlabelle — APPROVED** [review]

File: tests/sentry/stacktraces/test_functions.py (hunk 1 above)
Comment (Steelskin) [reply tests/sentry/stacktraces/test_functions.py]: Changed it to something more generic.

**Comment by JoshFerge** [comment]
thanks for the PR! looking at this now. will make sure it gets reviewed.

**Review by brustolin — COMMENTED** [review]
Everything seems good to me, but I think we should keep the closure number.

A function (specially in SwiftUI) may have many closures.
`closure #2 in MediaSlideshow.toSources` sounds more helpful

**Review by armenzg — APPROVED** [review]

**Comment by Steelskin** [comment]
> Everything seems good to me, but I think we should keep the closure number.
> 
> A function (specially in SwiftUI) may have many closures. `closure #2 in MediaSlideshow.toSources` sounds more helpful

The closure code is hard to parse, especially when we are dealing with closures in closures. Right now, on macOS, we just display `closure`. On Windows, we don't display the right closure number if we are in a closure in a closure. I feel like the current PR is an improvement for both cases. And with #72336 fixed, we can have the full prototype on hover.
